src/sections/create-dataset/CreateDatasetContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react'
+import {
+  CreateDatasetFormFields,
+  FormSubmissionService,
+  FormValidationService,
+  FormValidationResult
+} from '../../dataset/domain/useCases/createDataset'
+
+interface FormContextInterface {
+  fields: CreateDatasetFormFields
+}
+
+const defaultFormState: CreateDatasetFormFields = {
+  createDatasetTitle: ''
+}
+
+interface CreateDatasetFormProps {
+  formValidationService: FormValidationService
+  formSubmissionService: FormSubmissionService
+}
+
+export enum SubmissionStatusEnums {
+  NotSubmitted = 'NotSubmitted',
+  IsSubmitting = 'IsSubmitting',
+  SubmitComplete = 'SubmitComplete',
+  Errored = 'Errored'
+}
+
+export function useCreateDatasetForm({
+  formValidationService,
+  formSubmissionService
+}: CreateDatasetFormProps) {
+  const [formState, setFormState] = useState<FormContextInterface>({
+    fields: defaultFormState
+  })
+  const [submissionStatus, setSubmissionStatus] = React.useState<SubmissionStatusEnums>(
+    SubmissionStatusEnums.NotSubmitted
+  )
+  const [formErrors, setFormErrors] = useState<
+    Record<keyof CreateDatasetFormFields, string | undefined>
+  >({ createDatasetTitle: undefined })
+
+  const handleCreateDatasetFieldChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target
+    setFormState((prevState) => ({
+      ...prevState,
+      fields: { ...prevState.fields, [name]: value }
+    }))
+  }
+
+  const handleCreateDatasetSubmit = (event: FormEvent<HTMLFormElement>) => {
```

Comment:
Here I have the same comment about moving the event listener to the CreateDataset component and exposing in the hook a submitDataset function than in the above function